pkg/workloadmeta/collectors/internal/kubeapiserver/deployments.go

```diff
@@ -0,0 +1,85 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package kubeapiserver
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+const deploymentStoreName = "deployments-store"
+
+func init() {
+	resourceSpecificGenerator[deploymentStoreName] = newDeploymentStore
+}
+
+func newDeploymentStore(ctx context.Context, cfg config.Config, wlm workloadmeta.Store, client kubernetes.Interface) (*cache.Reflector, *reflectorStore, error) {
+	if !cfg.GetBool("language_detection.enabled") {
+		return nil, nil, fmt.Errorf("language detection is enabled") // we might remove this if we want to use deployments info in unified service tagging
+	}
+	deploymentListerWatcher := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			return client.AppsV1().Deployments(metav1.NamespaceAll).List(ctx, options)
```

Comment:
We can watch only deployment with the interesting labels. so we don't watch all the deployment. WDYT?

Reply:
I added a filter object in the reflectorStore. It's optional and can be used to filter deployments.